src/bin/mdbook-i18n.rs

```diff
@@ -0,0 +1,110 @@
+use mdbook::renderer::RenderContext;
+use serde::Deserialize;
+use std::collections::BTreeMap;
+use std::io;
+
+/// Parameters for the i18n renderer.
+///
+/// They are set in the `output.i18n` section of the book's `book.toml` file.
+#[derive(Deserialize)]
+struct I18nConfiguration {
+    /// A map of language codes to language names.
+    ///
+    /// ## Example
+    ///
+    /// ```toml
+    /// [output.i18n.languages]
+    /// "en" = "English"
+    /// "es" = "Spanish (Español)"
+    /// "ko" = "Korean (한국어)"
+    /// "pt-BR" = "Brazilian Portuguese (Português do Brasil)"
+    /// ```
+    #[serde(default)]
+    languages: BTreeMap<String, String>,
+    /// Default language code. It will not be translated.
+    default_language: Option<String>,
+
+    /// Whether to translate all languages or just the selected language, defaults to false.
+    #[serde(default)]
+    translate_all_languages: bool,
+    /// Whether to move the translations to the html directory, defaults to false.
+    ///
+    /// By default, translations's HTML output will live in `book/i18n/<language>/html`. If this is set to true, they will be moved to `book/html/<language>`.
```

Comment:
I think the `book/i18n/` directory here is wrong: mdbook will only create a per-renderer subdirectory if there are multiple renderers enabled. We should no assume that `html` or any other renderer is enabled.

Reply:
My suggestion would be to create a tiny blank book and experiement with the new renderer there. Don't use Comprehensive Rust as a starting point — since that book has a more complex setup than most (I think).